### PR TITLE
[Feature] Add Community Partners form

### LINKED ISSUE #194: [FEATURE] Add community partners form
### Detailed description

This was discuss on this PR https://github.com/web3community/devprotocol.xyz/pull/188#discussion_r765469316

Use this: https://tailwindui.com/components/marketing/sections/contact-sections#component-dec976a631662b509173ba8b848c49cd

### Context

It will allow our future partners to apply easily

### Possible implementation

The fields information will be

Founder Name
Community Name
Website
Email
Why do you want to be partner with us?
How did you hear about us?

Add radio button for Agreement to terms of use and code of conduct.

image

### Additional information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Simperfy**
#### Description of the change
- Added a modal with form in community partners that saves data on Google Sheets using [SheetDB](https://sheetdb.io/) with Basic Auth enabled. *(Thanks to @shuklaritvik06 for suggesting to use the SheetDB :heart:)*
- Sample CSV file can be found [here](https://docs.google.com/spreadsheets/d/1QrG1IkwAF9AMV84oEtRTWRRRahVRKNnt8dAJMRvaecM/edit?usp=sharing)
- Closes #194 

**NOTE: This PR needs to use `.env` variables which is currently not working on our version of astro. See #198**

```env
#sample .env
PUBLIC_SHEETDB_API_ID="abcdefghijklmnopqrstuvwxyz"
PUBLIC_SHEETDB_LOGIN="abcdefghijklmnopqrstuvwxyz"
PUBLIC_SHEETDB_PASSWORD="abcdefghijklmnopqrstuvwxyz"
```

#### Screenshots
Demo video *(I hard coded the credentials while making this demo)*:

https://user-images.githubusercontent.com/28738855/146648993-3f0f112d-2543-430b-b136-e28fd9c535d6.mp4

Proposed UI *(Desktop - light mode)*:
Screenshot from 2021-12-19 00-52-06

Proposed UI *(Desktop - dark mode)*:
Screenshot from 2021-12-19 00-48-50

Proposed UI *(Mobile - light mode)*:
Screenshot from 2021-12-19 00-53-44

Proposed UI *(Mobile - dark mode)*:
Screenshot from 2021-12-19 00-53-55

<hr>

**UPDATED** the left panel:
image

#### Checklist

**I agree to the following :-**

* [x]   Added description of the change
* [x]   I've read the [contributing guidelines](https://github.com/web3community/devprotocol.xyz/blob/main/CONTRIBUTING.md)
* [x]   Search previous suggestions before making a new PR, as yours may be a duplicate.
* [x]   I acknowledge that all my contributions will be made under the project's license.

Notes (**optional**): 

<a href="https://gitpod.io/#https://github.com/web3community/devprotocol.xyz/pull/199"></a>

### CODE REVIEW

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (Simperfy) [inline src/components/CommunityPartners/CommunityPartners.vue:192]: If you want to test this now, you must hard code this value. You can get it by creating a Google Form and creating an api with https://sheetdb.io. Don't forget to enable basic auth in the settings.

**Review by vinzvinci — COMMENTED** [review]
Wow! @Simperfy OP 🔥

**Comment by shuklaritvik06** [comment]
UI 🔥

**Comment by mkubdev** [comment]
Nice work @Simperfy ! 👍🏻

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartners.vue:137]:
```suggestion
          image: "https://github.com/github.png",
```

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartners.vue:168]:
```suggestion
          name: "Intlify",
```

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartners.vue:239]: Don't you think we should use `Your name` here?
Imagine I submit this form, but I'm not the founder. 😃

```suggestion
          'Your Name': founderName,
```

Not sure if I should change the variable; you might need to change that in other parts of the code. 😅

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartners.vue:260]: Missing an endline.
```suggestion
</script>

```

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartnersModal.vue:63]:
```suggestion
                <p>hi@devprotocol.xyz</p>
```

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartnersModal.vue:80]:
```suggestion
                  <span class="font-bold">Error!</span> Something went wrong, please try again.
```

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartnersModal.vue:91]:
```suggestion
                  <label for="founderName" class="community-partners-label">Your Name</label>
```

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartnersModal.vue:206]: Missing an endline.
```suggestion
</style>

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
I know this is a WIP PR, but I wanted to leave some small comments and suggestions. :)

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartners.vue:239]: I think we should ask @vinzvinci I just followed what he wrote in #194.

Comment (vinzvinci) [reply src/components/CommunityPartners/CommunityPartners.vue:240]: > I think we should ask @vinzvinci, I just followed what he wrote in #194.

Founder Name is best.

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (krishguptadev) [reply src/components/CommunityPartners/CommunityPartnersModal.vue:87]: It is good as is

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (krishguptadev) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: Update: I've updated the version in this PR.

Comment (vinzvinci) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: I already created it, does the google sheet need to be public or is it okay if I make it private?

Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: It works whether it's public or private.

Comment (vinzvinci) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: > It works whether it's public or private.

Okay I set it to private

Comment (vinzvinci) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: I need to put this on the environment variable right?

- The spreadsheet URL
- auth login and auth password

or only auth login and pass?

Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: You need three things:
- address (a.k.a API ID)
- login
- password

The address can be found here: https://sheetdb.io/apis
Screenshot 2021-12-20 072808

The login and password can be found in the settings:
Screenshot 2021-12-20 073011

Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: I should probably rename the variable `address` to `api_id` to avoid confusion.

Comment (vinzvinci) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: > Yes you need to put them in the environment variables.
> 
> You need three things:
> 
> * address (a.k.a API ID)
> * login
> * password

Cool thank you 🎉 

> I should probably rename the variable `address` to `api_id` to avoid confusion.

Hehehe yes

Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: > Update: I've updated the version in this PR.

Thanks but the tests fails due to the builds failing.

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (vinzvinci) [inline src/components/CommunityPartners/CommunityPartnersModal.vue:56]: add a hyperlink for support

`<a href="https://discord.gg/VwJp4KM">support</a>`

Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartnersModal.vue:56]: Working on it. 💯

Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartnersModal.vue:56]: I'll push the changes once krishguptadev fixed the failing tests.

Comment (vinzvinci) [reply src/components/CommunityPartners/CommunityPartnersModal.vue:56]: Okay thanks 💯

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (vinzvinci) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: CC @krishguptadev

Comment (krishguptadev) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: It should be on localhost:3000/devprotocol.xyz/community-partner due to the next update 👍

Comment (krishguptadev) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: Update: I've added a patch to make it appear or /community-partners

Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartners.vue:193]: Awesome 💯

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartners.vue:244]:
```suggestion
          'Why do you want to partner with us?': whyPartnerWithUs,
```

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartnersModal.vue:56]:
```suggestion
              <p class="text-lg">Have any questions/comments, or want more information? Contact any of our team members via <a href="https://discord.gg/VwJp4KM">Discord</a>.</p>
```

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (Panquesito7) [inline src/components/CommunityPartners/CommunityPartnersModal.vue:103]:
```suggestion
                  <label for="whyPartnerWithUs" class="community-partners-label">Why do you want to partner with us?</label>
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (vinzvinci) [reply src/components/CommunityPartners/CommunityPartnersModal.vue:56]: I think this is better?
 'Have any questions/comments? Reach out to our team members via discord'

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
For the Discord logo, please take it (the best color like white/black) from the official Discord page https://discord.com/branding

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartners.vue:244]: @vinzvinci You need to change the header column in the google forms according to this one.

From:
image

To:
image

Let me know once you read this before I commit so we can avoid bugs.

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (Panquesito7) [reply src/components/CommunityPartners/CommunityPartnersModal.vue:56]: Sounds good!

Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartnersModal.vue:56]: Yeah, less word = better.

**Comment by Simperfy** [comment]
> For the Discord logo, please take it (the best color like white/black) from the official Discord page https://discord.com/branding

Oh I didn't know such page exists. I'll use this, thanks ❤️

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (Simperfy) [reply src/components/CommunityPartners/CommunityPartnersModal.vue:56]: Here's how it will look:
image

What do you guys think?

**Review by Panquesito7 — APPROVED** [review]
Amazing work! 🚀 Thanks so much for contributing! 😄👍
Let's wait for all the team to review this PR. 🙂

**Review by krishguptadev — DISMISSED** [review]

**Review by krishguptadev — COMMENTED** [review]
Awesome! This is a approval from my side, I removed my previous approval so that I can turn on auto-merge

**Comment by Panquesito7** [comment]
Please don't enable auto-merge until all the team approves. :)

**Comment by krishguptadev** [comment]
> Please don't enable auto-merge until all the team approves. :)

You've given approval, I've given approval, only vinz is left, and auto-merge will merge when vinz approves

File: src/components/CommunityPartners/CommunityPartnersModal.vue
Comment (shuklaritvik06) [reply src/components/CommunityPartners/CommunityPartnersModal.vue]: looks good

**Review by vinzvinci — COMMENTED** [review]
Hello, can you check if the navbar and its sub-nav-item are showing? On my device, it doesn't show.

**Comment by shuklaritvik06** [comment]
> Hello, can you check if the navbar and its sub-nav-item are showing? On my device, it doesn't show.

In the navbar the dropdown is not working and also that dark mode one , I thought it was just for me 😅

**Comment by Simperfy** [comment]
> Hello, can you check if the navbar and its sub-nav-item are showing? On my device, it doesn't show.

Hmm.. Thats weird, the navbar works on localhost but on netlify it doesn't.

**Comment by Panquesito7** [comment]
> > Please don't enable auto-merge until all the team approves. :)
> 
> You've given approval, I've given approval, only vinz is left, and auto-merge will merge when vinz approves

image

**Review by vinzvinci — DISMISSED** [review]
LGTM 👍 I tested it on my own and it really works. We're using vercel at the [GitHub Org. of Dev Protocol](https://github.com/dev-protocol), I think it also works there.

File: src/components/CommunityPartners/CommunityPartners.vue
Comment (vinzvinci) [reply src/components/CommunityPartners/CommunityPartners.vue]: Hey I remove that **be** on the spreadsheet

**Pull request merged**
